src/main/java/spireCafe/interactables/attractions/punchingbag/PunchingBagAttraction.java

```diff
@@ -0,0 +1,249 @@
+        }
+
+        int cardMulti = 1;
+        for (AbstractCard c : deck) {
+            int cardScore = 0;
+            switch (c.rarity) {
+                case COMMON:
+                    cardMulti = 2;
+                    break;
+                case UNCOMMON:
+                    cardMulti = 4;
+                    break;
+                case RARE:
+                case SPECIAL:
+                    cardMulti = 8;
+                    break;
+                default:
+                    cardMulti = 1;
+                    break;
+            }
+            switch (c.type) {
+                case ATTACK:
+                    cardScore += (cardMulti * B_ATK) + strength;
+                    attackCount += 1;
+                    break;
+                case SKILL:
+                    cardScore += cardMulti * B_SKL;
+                    skillCount += 1;
+                    break;
+                case POWER:
+                    if (c.rarity.equals(CardRarity.RARE)) {
+                        multiplier *= R_PWR;
+                    } else if (c.rarity.equals(CardRarity.UNCOMMON)) {
+                        multiplier *= U_PWR;
+                    } else {
+                        multiplier *= C_PWR;
+                    }
+                    break;
+                case CURSE:
+                case STATUS:
+                    if (!c.cardID.equals(AscendersBane.ID)) {
+                        multiplier *= CURSE;
+                    }
+                    break;
+                default:
+                    break;
+            }
+
+            // Additive bonuses
+            if (c.hasTag(CardTags.STRIKE) && p.hasRelic(StrikeDummy.ID)) {
+                score += 2;
+            }
+            if (c.cardID.equals(Claw.ID)){
+                cardScore += (2 + strength) * clawCount;
+                clawCount += 1;
+            }
+            if (p.hasRelic(Shuriken.ID) && c.type.equals(CardType.ATTACK) && attackCount % 3 == 0) {
+                strength += 1;
+            }
+            
+            // Multiplier bonuses
+            cardScore *= Math.pow(1.5F, c.timesUpgraded);
+
+            if (p.hasRelic(PenNib.ID) && c.type.equals(CardType.ATTACK) && attackCount % 10 == 0) {
+                cardScore *= 2;
+            }
+            score += cardScore;
+        }
+
+        // Additive bonuses (Total)
+        if (p.hasRelic(Akabeko.ID) && attackCount > 0) {
+            score += 8;
+        }
+        if (p.hasRelic(LetterOpener.ID) && skillCount > 0) {
+            score += 5 * (skillCount / 3);
+        }
+
+        return (int)(score * multiplier);
+    }
+
+    private AttackEffect getAttackFX() {
+        AttackEffect[] slashFX = AbstractDungeon.player.getSpireHeartSlashEffect();
+        return slashFX[MathUtils.random(slashFX.length - 1)];
+    }
+
+    private void saveHiScore(int score) {
+        try {
+            SpireConfig config = new SpireConfig(Anniv7Mod.modID, "anniv7Config");
```

Comment:
Hmm, not sure if creating a config with the same ID as the one we create in the main mod file is a good idea, especially since we already have an instance save there under ` modConfig`, might be better to use that since I'm unsure how it deals with being instanciated twice.